ls8/cpu.c

```diff
@@ -83,28 +66,36 @@ void alu(struct cpu *cpu, enum alu_op op, unsigned char regA, unsigned char regB
   unsigned char b = cpu->reg[regB];
 
   switch (op) {
-    case ALU_MUL:
-      cpu->reg[regA] = a * b; //multiply the values in two registers together and store the result in registerA.
+    case ALU_AND:
+      cpu->reg[regA] = cpu->reg[regA] & cpu->reg[regB];
       break;
 
+    case ALU_OR:
+    cpu->reg[regA] = cpu->reg[regA] | cpu->reg[regB];
+    break;
+
+    case ALU_MUL:
+    cpu->reg[regA] = a * b; //multiply the values in two registers together and store the result in registerA.
+    break;
+
     case ALU_ADD:
     cpu->reg[regA] = a + b; //add the value in two registers and store the result in registerA.
     break;
 
-    // case ALU_CMP: //Compare the values in two registers. FL bits: 00000LGE
-    // if (a < b) //L Less-than: during a CMP, set to 1 if registerA is less than registerB, zero otherwise.
-    // {
-    //   cpu->FL = 0b00000100;
-    // }
-    // else if (b > a) //G Greater-than: during a CMP, set to 1 if registerA is greater than registerB, zero otherwise.
-    // {
-    //   cpu->FL = 0b00000010;
-    // }
-    // else //E Equal: during a CMP, set to 1 if registerA is equal to registerB, zero otherwise.
-    // {
-    //   cpu->FL = 0b00000001;
-    // }
-    // break;
+    case ALU_CMP: //Compare the values in two registers. FL bits: 00000LGE
+    if (a < b) //L Less-than: during a CMP, set to 1 if registerA is less than registerB, zero otherwise.
+    {
+      cpu->FL = 0b00000100;
+    }
+    else if (b > a) //G Greater-than: during a CMP, set to 1 if registerA is greater than registerB, zero otherwise.
+    {
+      cpu->FL = 0b00000010;
+    }
+    else //E Equal: during a CMP, set to 1 if registerA is equal to registerB, zero otherwise.
+    {
+      cpu->FL = cpu->FL| 0b00000001;
```

Comment:
Why the choice to to `=|` here? Seems like that could lead to inaccurate results. What if cpu->FL starts off as `0b00000010` or `0b00000100`?